### PR TITLE
async-h1 update for http-types 2.0.0

### PR DESCRIPTION
**Pull request by jbr**

### CODE REVIEW

**Comment by jbr** [comment]
@yoshuawuyts Aside from the fact that this is building against github master, this is ready for review

File: src/chunked/decoder.rs [hunk 1]
```diff
@@ -583,20 +580,30 @@ mod tests {
...
+
+            assert_eq!(as_vec.len(), 1);
+            let (header_name, header_values) = as_vec[0];
+            assert_eq!(
+                header_name,
+                &"Expires"
+                    .parse::<http_types::headers::HeaderName>()
+                    .unwrap()
+            );
+            assert_eq!(header_values.iter().collect::<Vec<_>>().len(), 1);
```
Comment (yoshuawuyts) [inline src/chunked/decoder.rs:601]: I believe we should be able to do much simpler assertions here

File: src/chunked/decoder.rs [hunk 2]
```diff
@@ -583,20 +580,30 @@ mod tests {
...
             assert_eq!(
-                trailer.iter().collect::<Vec<_>>(),
-                vec![(
-                    &"Expires".parse().unwrap(),
-                    &vec!["Wed, 21 Oct 2015 07:28:00 GMT".parse().unwrap()],
-                )]
+                header_values.last(),
+                &"Wed, 21 Oct 2015 07:28:00 GMT"
+                    .parse::<http_types::headers::HeaderValue>()
+                    .unwrap()
```
Comment (yoshuawuyts) [inline src/chunked/decoder.rs:606]: I don't believe we need to parse and unwrap here

File: src/server/encode.rs [hunk 3]
```diff
@@ -5,7 +5,10 @@ use std::pin::Pin;
 use async_std::io;
 use async_std::io::prelude::*;
 use async_std::task::{Context, Poll};
-use http_types::Response;
+use http_types::{
+    headers::{CONTENT_LENGTH, DATE, TRANSFER_ENCODING},
+    Response,
+};
```
Comment (yoshuawuyts) [inline src/server/encode.rs:11]: Can we split this into two separate imports?

File: src/chunked/decoder.rs (hunk 1 above)
Comment (jbr) [reply src/chunked/decoder.rs]: I _think_ this is resolved, but I also didn't do a particularly thorough reworking of any of these tests, just got them to compile and pass

**Pull request merged**